cumulusci/tasks/push/pushfails.py

```diff
@@ -0,0 +1,69 @@
+""" simple task(s) for reporting on push upgrade jobs.
+
+this doesn't use the nearby push_api module, and was just a quick ccistyle
+get the job done kinda moment.
+"""
+
+import re
+import csv
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+
+
+class PushFailTask(BaseSalesforceApiTask):
+    """ Produce a report of the failed and otherwise anomalous push jobs from
+    a specified push request id, and write it out to a results CSV.  """
+
+    task_doc = __doc__
+    task_options = {
+        "request_id": {
+            "description": "PackagePushRequest ID for the request you need to report on.",
+            "required": True,
+        },
+        "result_file": {"description": "Where to write results"},
+    }
+    api_version = "43.0"
+    query = "SELECT ID, SubscriberOrganizationKey, (SELECT ErrorDetails, ErrorMessage, ErrorSeverity, ErrorTitle, ErrorType FROM PackagePushErrors) FROM PackagePushJob WHERE PackagePushRequestId = '{request_id}' AND Status !='Succeeded'"
+    gack = re.compile(r"error number: ([\d-]+) \(([\d-]+)\)")
+    headers = [
+        "OrganizationId",
+        "ErrorSeverity",
+        "ErrorTitle",
+        "ErrorType",
+        "ErrorMessage",
+        "Gack Id",
+        "Stacktrace Id",
+    ]
+
+    def _run_task(self):
+        formatted_query = self.query.format(**self.options)
+        self.logger.debug("Running query: " + formatted_query)
```

Comment:
Nothing to worry about now, but we should probably build this into simple-salesforce or a wrapper thereof.

Reply:
Actually I have some MIT-licensed code I wrote for a former client that does this, as well as automatically retrying calls if the SF session expired.

Reply:
I was thinking about a generic BaseQueryTask that gets subclassed like:
```
class PushReportFail(BaseQueryTask):
    task_doc = "<blah>"
    query = "<blah> {using_options}"
    def _query_results(self, result):
        pass
```
and maybe even prints to stdout by default? later tho...

Reply:
But then we're assuming that we only do queries from tasks, that the task does a single query and that it doesn't do other things apart from querying. I think it's better decoupling to build the logging into a query function that can be used from wherever.

Reply:
I guess we only really have an API to use within the context of a task though, so the first of those assumptions is reasonable.

Reply:
some wrapper thereof is definitely an ineresting idea...